### PR TITLE
Added SQLToCollection and DBSession#collection

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

**Comment by gakuzzzz** [comment]
Sorry, I forget that in my PR there is one discuss point.

Do `SQL#toCollection` and `SQL#toCollection` require 	parentheses?
It have no side-effect. however it lose the consistency other methods.

**Comment by seratch** [comment]
Is collection the best name at all? Scala doesn't have such a type.

**Comment by gakuzzzz** [comment]
the method can build any types that has `CanBuildFrom`.
So, It is rather better that not-exists type name.

**Comment by seratch** [comment]
I'm just wondering collection is the best name althogh it's reasonable enough.
In this case, I'm afraid that #toCollection may look strange.

**Comment by gakuzzzz** [comment]
I see.
the similar method in Traversable is `to`.

```scala
withSQL {
  ...
}.map(Foo as f).to[Vector]()
```

What do you think?

**Comment by seratch** [comment]
>withSQL { ... }.map(Foo as f).to\[Vector\]()

I think it's too versatile. Library users will expect any types are acceptable as the type parameter of #to method. 

hm, should we go with `collection` after all?

File: scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala [hunk 1]
```diff
@@ -247,7 +249,21 @@ trait DBSession extends LogSupport with LoanPattern {
...
+   * Returns query result as any Collection object.
+   *
+   * @param template SQL template
+   * @param params parameters
+   * @param extract extract function
+   * @tparam A return type
+   * @tparam C[_] return collection type
+   * @return result as C[A]
+   */
+  def collection[A, C[_]](template: String, params: Any*)(extract: WrappedResultSet => A)(implicit cbf: CanBuildFrom[Nothing, A, C[A]]): C[A] = {
```
Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: The unfortunate Scala real. We can't help it.

File: scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala [hunk 2]
```diff
@@ -247,7 +249,21 @@ trait DBSession extends LogSupport with LoanPattern {
...
+  }
+
+  /**
+   * Returns query result as any Collection object.
+   *
+   * @param template SQL template
+   * @param params parameters
+   * @param extract extract function
+   * @tparam A return type
+   * @tparam C[_] return collection type
```
Comment (xuwei-k) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: Is this correct scaladoc syntax?

Comment (seratch) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: I'm not sure about this. ping @gakuzzzz

Comment (gakuzzzz) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: I can not find the scaladoc spec of the higher kinded tparam.
I'll try generate scaladoc and check the behavior at home.

Comment (gakuzzzz) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: This is invalid scaladoc syntax.
It should be `@tparam C return collection type`

Comment (xuwei-k) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: fixed

Comment (gakuzzzz) [inline scalikejdbc-core/src/main/scala/scalikejdbc/DBSession.scala]: thanks!!

**Pull request merged**